cmd/extract.go

```diff
@@ -0,0 +1,108 @@
+package cmd
+
+import (
+	"os"
+	"io"
+	"fmt"
+
+	"github.com/sector-f/jh_extract/nvc"
+)
+
+func extractNVC(arcPath string, pathlist []string, outputDirectory string, extractUnknown bool) error {
+	r, err := os.Open(arcPath)
+	if err != nil {
+		return err
+	}
+
+	hashToPath := map[nvc.Hash]string{}
+	for _, p := range pathlist {
+		hash := nvc.String2Hash(p)
+		hashToPath[hash] = p
+	}
+
+	var tocEntries []nvc.TocEntry
+	tocEntries, err = nvc.ReadToc(r)
+	if err != nil {
+		return err
+	}
+
+	entries := map[nvc.Hash]nvc.TocEntry{}
+	for _, e := range(tocEntries) {
+		entries[e.Hash] = e
+	}
+
+	extractedCount := 0
+
+	for hash, nvcEntry := range entries {
+		switch nvcEntry.Flags {
+		case nvc.Uncompressed:
+			// Do nothing
+		case nvc.Compressed:
+			// TODO: Perform zlib decompression
+		default:
+			continue
+		}
```

Comment:
This is the wrong approach if we're going to treat these as bitmask flags (i.e. encrypted is 2 rather than 3).

Should probably have something like

```go
var (
    compressed bool
    encrypted  bool
)
```

And then do some bit-twiddling to assign the correct values to those variables (and then act accordingly)

Reply:
Can we change the value of `nvc.Encrypted` back to 3 for now then?

Reply:
woops fixed i